common/v2/features/SendAssets/components/SignTransactionWallets/Metamask.tsx

```diff
@@ -27,4 +98,61 @@ export default class SignTransactionMetaMask extends Component {
       </div>
     );
   }
+  // private signTransaction() {
+
+  // }
+
+  private initMetMaskPolling() {
+    setInterval(this.getMetaMaskProviderAndSigner, 2000);
+  }
+
+  private checkAddressMatches(metaMaskAddress: string, stateValues: ISendState) {
+    const desiredAddress = utils.getAddress(stateValues.transactionFields.senderAddress);
+
+    if (metaMaskAddress === desiredAddress) {
+      return this.setState({ metaMaskAccountMatches: true });
+    } else {
+      return this.setState({ metaMaskAccountMatches: false });
+    }
+  }
+
+  // private checkNetworkMatches(metaMaskNetwork: string, stateValues: ISendState) {
+  //   const desiredNetwork = stateValues.network;
+
+  //   if (metaMaskNetwork === desiredNetwork) {
+  //     return this.setState({ metaMaskNetworkMatches: true });
+  //   } else {
+  //     return this.setState({ metaMaskNetworkMatches: false });
+  //   }
+  // }
+
+  private async getMetaMaskProviderAndSigner() {
+    if (window.ethereum) {
+      await window.ethereum.enable();
+      const metaMaskProvider = new ethers.providers.Web3Provider(window.ethereum);
+      const metaMaskSigner = await metaMaskProvider.getSigner();
+      const metaMaskNetwork = await metaMaskProvider.getNetwork();
+      const metaMaskAddress = await metaMaskSigner.getAddress();
+
+      const checksumAddress = utils.getAddress(metaMaskAddress);
+
+      await this.setState({
+        account: checksumAddress,
+        network: metaMaskNetwork.chainId,
+        metaMaskProvider,
+        metaMaskSigner
+      });
+
+      this.watchForAccountChanges(window.ethereum);
+      this.checkAddressMatches(checksumAddress, this.props.stateValues);
+      console.log(this.state);
+    } else {
+      const defaultProvider = ethers.getDefaultProvider(DEFAULT_NETWORK_FOR_FALLBACK);
+      this.setState({ provider: defaultProvider });
+    }
+  }
+
+  private watchForAccountChanges(ethereum: NonNullable<Window['ethereum']>) {
+    ethereum.on('accountsChanged', this.getMetaMaskProviderAndSigner);
```

Comment:
We don't need to reconnect to the network on each accountChange.
We can simplify by only getting the address, thus removing the recursive call.